src/XRP/model/xrp-transaction.ts

```diff
@@ -67,39 +101,66 @@ export default class XrpTransaction {
         .map((signer) => XrpSigner.from(signer, xrplNetwork))
     }
 
-    const sourceTag = transaction.getSourceTag()?.getValue()
+    const signingPublicKey = transaction.getSigningPublicKey()?.getValue_asB64()
+    if (signingPublicKey == undefined) {
+      throw new XrpError(
+        XrpErrorType.MalformedProtobuf,
+        'Transaction protobuf is missing valid `signingPublicKey` field.',
+      )
+    }
+    if (!signers && signingPublicKey == '') {
+      throw new XrpError(
+        XrpErrorType.MalformedProtobuf,
+        'Transaction protobuf does not have a `signingPublicKey` field or a `signers` field.',
+      )
+    }
 
-    const sourceXAddress = XrpUtils.encodeXAddress(
-      account,
-      sourceTag,
-      xrplNetwork === XrplNetwork.Test,
-    )
+    const transactionSignature = transaction
+      .getTransactionSignature()
+      ?.getValue_asU8()
+    if (!transactionSignature) {
+      throw new XrpError(
+        XrpErrorType.MalformedProtobuf,
+        'Transaction protobuf is missing valid `transactionSignature` field.',
+      )
+    }
 
     let paymentFields
     let type
     switch (transaction.getTransactionDataCase()) {
       case Transaction.TransactionDataCase.PAYMENT: {
         const payment = transaction.getPayment()
         if (!payment) {
-          return undefined
+          throw new XrpError(
+            XrpErrorType.MalformedProtobuf,
+            'Transaction protobuf is type Payment but is missing `payment` field.',
+          )
         }
-        paymentFields = payment && XrpPayment.from(payment, xrplNetwork)
-        if (!paymentFields) {
-          return undefined
-        }
-        type = payment && XrpTransactionType.Payment
+        paymentFields = XrpPayment.from(payment, xrplNetwork)
+        type = XrpTransactionType.Payment
         break
       }
+      case Transaction.TransactionDataCase.TRANSACTION_DATA_NOT_SET: {
+        throw new XrpError(
+          XrpErrorType.MalformedProtobuf,
+          'Transaction protobuf is missing transaction data.',
```

Comment:
```suggestion
          'Transaction protobuf is missing transaction data case.',
```

Reply:
My understanding of how the protobuf object works is that the data case gets set automatically, based on which data it has. So if it has no data case, then it's because it has no data.

Reply:
I think you're right about that, but I added this word just to make it clear that it's not the "transaction" field that's missing, but rather the type... since this is mostly human-facing.  Either way works for me